test.lua

```diff
@@ -192,19 +192,18 @@ function max_events_test(N)
 
     local parser = lhp.request(cbs)
     local input = table.concat(input_tbl)
+    local result = parser:execute(input)
 
     N = N * 2
     if (#input == result) and ( N < 100000 ) then
         return max_events_test(N)
     end
 
-    local result = parser:execute(input)
-
     input = input:sub(result + 1)
 
     -- We should have generated a stack overflow event that should be
     -- handled gracefully... note that
-    ok(#input > result,
+    ok(#input < result,
```

Comment:
Why would 'result' be larger then the length of the input?

Reply:
I see why 'result' would be larger.  But why even keep this check?  shouldn't it always pass?

The point of that check before was to make sure the callbacks were still called and the parsing could be resumed (if the Lua code really wants to keep parsing the input).